### PR TITLE
[SPARK-20945] Fix TID key not found in TaskSchedulerImpl

### PR DESCRIPTION
**Pull request by liupc**
## What changes were proposed in this pull request?

This pull request fix the TaskScheulerImpl bug in some condition.
Detail see:
https://issues.apache.org/jira/browse/SPARK-20945

(Please fill in changes proposed in this fix)

## How was this patch tested?
manual tests
(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by JoshRosen** [comment]
Hi @liupc,

Thanks for the JIRA and PR. Could you edit the PR's title to incorporate the JIRA number? e.g.

> [SPARK-20945] Fix TID key not found in TaskSchedulerImpl

Also, could you re-target this PR to be against the `master` branch? It looks like you've opened it against `branch-2.1`. Our workflow for these kind of bugs is generally to open PRs against the master branch: committers will take care of backporting your fix to appropriate maintenance branches.

File: core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala [hunk 1]
```diff
@@ -240,8 +240,10 @@ private[spark] class TaskSchedulerImpl private[scheduler](
         // 2. The task set manager has been created but no tasks has been scheduled. In this case,
         //    simply abort the stage.
         tsm.runningTasksSet.foreach { tid =>
-          val execId = taskIdToExecutorId(tid)
-          backend.killTask(tid, execId, interruptThread, reason = "stage cancelled")
+          if (taskIdToExecutorId.contains(tid)) {
+            val execId = taskIdToExecutorId(tid)
+            backend.killTask(tid, execId, interruptThread)
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala]: I think you lost the value of "reason" here.
It's not a big deal but you could also write:
```
taskIdToExecutorId.get(tid).foreach(execId => backend.killTask(tid, execId, interruptThread, reason = "stage cancelled"))

Comment (liupc) [inline core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala]: Looks good! I will update soon, thanks!

**Comment by srowen** [comment]
Merged to master

**Comment by jinxing64** [comment]
Why this is not merged into 2.2 ?

**Pull request closed without merging**